sdk/src/main/scala/com/horizen/storage/SidechainStateStorage.scala

```diff
@@ -44,30 +53,56 @@ class SidechainStateStorage (storage : Storage, sidechainBoxesCompanion: Sidecha
     }
   }
 
-  def update(version : ByteArrayWrapper, boxUpdateList : Set[SidechainTypes#SCB],
+  def update(version : ByteArrayWrapper, withdrawalEpochInfo: WithdrawalEpochInfo,
+             boxUpdateList : Set[SidechainTypes#SCB],
              boxIdsRemoveList : Set[Array[Byte]]) : Try[SidechainStateStorage] = Try {
     require(boxUpdateList != null, "List of Boxes to add/update must be NOT NULL. Use empty List instead.")
     require(boxIdsRemoveList != null, "List of Box IDs to remove must be NOT NULL. Use empty List instead.")
     require(!boxUpdateList.contains(null), "Box to add/update must be NOT NULL.")
     require(!boxIdsRemoveList.contains(null), "BoxId to remove must be NOT NULL.")
+    require(withdrawalEpochInfo != null, "WithdrawalEpochInfo must be NOT NULL.")
 
     val removeList = new JArrayList[ByteArrayWrapper]()
     val updateList = new JArrayList[JPair[ByteArrayWrapper,ByteArrayWrapper]]()
 
     for (r <- boxIdsRemoveList)
       removeList.add(calculateKey(r))
 
-    for (b <- boxUpdateList)
+    for (b <- boxUpdateList.filter(box => !box.isInstanceOf[WithdrawalRequestBox]))
       updateList.add(new JPair[ByteArrayWrapper, ByteArrayWrapper](calculateKey(b.id()),
         new ByteArrayWrapper(sidechainBoxesCompanion.toBytes(b))))
 
+    if (boxUpdateList.count(box => box.isInstanceOf[WithdrawalRequestBox]) > 0) {
```

Comment:
I don't think that storage should do the withdrawal requests special calculation logic.
State itself should calculate box changes -> extract requests -> get from storage actual epoch info and requests list -> calculate boxes to add (remove requests), boxes to remove, new epoch info value, withdrawals to save for calculated epoch info.